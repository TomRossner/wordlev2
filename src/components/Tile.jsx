import React from 'react';

const Tile = ({tile}) => {

  return (
    <div className={`tile ${tile.colorClass}`}>
        {tile.letter}
    </div>
  )
}

export default Tile;