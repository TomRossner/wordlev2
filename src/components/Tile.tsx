import React from 'react';
import { ITile } from '../interfaces';

interface TileProps {
  tile: ITile;
  handleRemoveShake: () => void; 
}

const Tile: React.FC<TileProps> = ({tile, handleRemoveShake}) => {

  return (
    <div className={`${tile.letter ? 'tile filled' : 'tile'} ${tile.colorClass}`} onAnimationEnd={handleRemoveShake}>
        {tile.letter.toUpperCase()}
    </div>
  )
}

export default Tile;