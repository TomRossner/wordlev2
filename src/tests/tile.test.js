import renderer from 'react-test-renderer';
import Tile from '../components/Tile';

const tileMock = {
    letter: 'a',
    colorClass: 'gray'
}

it('Renders correctly', () => {
    const tile = renderer.create(<Tile tile={tileMock}/>).toJSON();

    expect(tile).toMatchSnapshot();
})