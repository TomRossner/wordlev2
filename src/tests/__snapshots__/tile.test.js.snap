// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="tile filled gray"
>
  A
</div>
`;
